src/server/req_modify.c

```diff
@@ -885,6 +888,15 @@ req_modifyReservation(struct batch_request *preq)
 
 		psatl = (svrattrl *)GET_NEXT(psatl->al_link);
 	}
+
+	presv->ri_alter_state = presv->ri_wattr[RESV_ATR_state].at_val.at_long;
```

Comment:
Instead of setting this just to reset it in the above error cases, move this to the top of the while loop.  There is only one error case you don't call revert_alter_resv_times().  You can just clear this in that reject case.

Also, I'd like to see if we can combine this check with its counterpart in the running case above.  Instead of checking it up there, just do it once down here for both cases.

Reply:
We will have to keep the check to see if the reservation is in running state that there are no jobs in it. Otherwise it will execute the start time case in the while loop and presv->ri_alter_stime changes. This causes a problem with standing reservations with already altered start time. For eg this test will fail
1) Submit a standing reservation
2) Alter start time of the reservation after it starts
3) Submit a job to it and while the job is running alter start time again, this fails
4) After the reservation occurance ends, the start time of next occurance remains what it was altered too in step 2 and not the original one